modules/apps/frontend-js/frontend-js-clay-sample-web/src/main/resources/META-INF/resources/view.jsp

```diff
@@ -16,8 +16,32 @@
 
 <%@ include file="/init.jsp" %>
 
-<div>
-	<react:component
-		module="js/App"
-	/>
+<div class="row">
+	<div class="col">
+		<div>
+			<react:component
+				module="js/App"
+			/>
+		</div>
+	</div>
+
+	<div class="col">
+		<h2>JSP!</h2>
+
+		<div id="jsp_age">Age:</div>
+
+		<input id="input_name" placeholder="Update Name" />
+
+		<aui:script>
+			Liferay.State.subscribe('test', () => {
+				const state = Liferay.State.get('test').store.getState();
+	
+				document.getElementById('jsp_age').innerText = 'Age: ' + state.age;
+			});
+
+			document.getElementById('input_name').addEventListener('input', function(event) {
+				Liferay.State.get('test').store.dispatch({type: 'UPDATE_NAME', value: event.target.value})
+			});
+		</aui:script>
+	</div>
```

Comment:
This is just for demo purposes, it really wouldn't matter if these two implementations lived in entirely different portlets or not.